### PR TITLE
Record swap fee amount in Recovery Mode

### LINKED ISSUE #1092: Review total swap fees when in recovery mode
[Cantina 268](https://cantina.xyz/code/949ad7c5-ea14-427d-b10a-54e33cef921b/findings/268?with_events=false&limit=20&status=new%2Cdisputed%2Cduplicate%2Cconfirmed%2Cacknowledged%2Cfixed) (informational)

### PR DESCRIPTION
**Pull request by EndymionJkb**
# Description

The raw swap fee amount should be emitted in the Swap event, even in Recovery Mode. This is a true edge case, since normally pools in Recovery Mode would also be paused, but in case Recovery Mode was enabled separately - and swaps still worked - the Swap event will now report it. (Even though protocol and pool creator fees aren't charged in recovery mode, swap fees still apply.)

Credits to @dmitriia for the finding.

## Type of change

- [X] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [X] The diff is legible and has no extraneous changes
- [X] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [X] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

Resolves #1092

### CODE REVIEW

File: pkg/vault/contracts/Vault.sol [hunk 1]
```diff
@@ -1039,33 +1039,36 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
...
             // modify it. Given that all of the fee may belong to the pool creator (i.e. outside pool balances),
             // we round down to protect the invariant.
+
             totalSwapFeeAmountRaw = totalSwapFeeAmountScaled18.toRawUndoRateRoundDown(
                 poolData.decimalScalingFactors[index],
                 poolData.tokenRates[index]
             );
 
-            uint256 aggregateSwapFeePercentage = poolData.poolConfigBits.getAggregateSwapFeePercentage();
+            if (poolData.poolConfigBits.isPoolInRecoveryMode() == false) {
```
Comment (jubeira) [inline pkg/vault/contracts/Vault.sol:1052]: Let's leave a small comment, something like "aggregate fees are not charged in recovery mode, but we still return the total swap fee raw to report offchain.

**Review by jubeira — APPROVED** [review]
Great job, LGTM!

Thanks again @dmitriia! :raised_hands:

File: pkg/vault/test/foundry/VaultSwap.t.sol [hunk 2]
```diff
@@ -234,6 +234,36 @@ contract VaultSwapTest is BaseVaultTest {
         );
     }
 
+    function testSwapEventExactInRecovery() public {
```
Comment (jubeira) [inline pkg/vault/test/foundry/VaultSwap.t.sol:237]: This also happens for add / remove :/

Should we add a small test too?

**Review by jubeira — COMMENTED** [review]
One more late comment.

File: pkg/vault/test/foundry/VaultSwap.t.sol (hunk 2 above)
Comment (EndymionJkb) [reply pkg/vault/test/foundry/VaultSwap.t.sol:237]: That is harder. I tweaked some tests in VaultLiquidity.t.sol to set fees and unbalance, and verified in the logs that it was reporting the fee in PoolBalanceChanged, but it's hard to predict the values (rounding, etc.).

I think I'm out of time for that tonight; want to finish the rest of it.

**Review by elshan-eth — APPROVED** [review]
Cool! LGTM!

**Review by joaobrunoah — APPROVED** [review]
LGTM!

**Pull request merged**